clerk/querier.go

```diff
@@ -46,3 +50,40 @@ func handlerQueryRecord(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) (
 	}
 	return bz, nil
 }
+
+// HandlerQueryStateSyncer will select next 3 state syncers
+func HandlerQueryStateSyncer(ctx sdk.Context, req abci.RequestQuery, clerkKeeper Keeper, stakingKeeper staking.Keeper) ([]byte, sdk.Error) {
+
+	// Total no of state sync events
+	eventCount := clerkKeeper.GetStateSyncEventCount(ctx)
+
+	// Total no of validators
+	validatorCount := len(stakingKeeper.GetAllValidators(ctx))
+	stateSyncerList := []types.Validator{}
+
+	// No of state syncers to return
+	syncerCount := 3
+
+	// if no of validators is less than three, return existing validators
+	if validatorCount < 3 {
+		syncerCount = validatorCount
+	}
+
+	// Select next 3 active validators
+	for i := 0; i < syncerCount; {
+		valIndex := eventCount % uint64(validatorCount)
+		validator, _ := stakingKeeper.GetValidatorFromValID(ctx, types.ValidatorID(valIndex+1))
+		if stakingKeeper.IsCurrentValidatorByAddress(ctx, validator.Signer.Bytes()) {
```

Comment:
Can't we get validators from current validator set instead of getting all validators and filter by validator status?

Reply:
Yeah. we can do that. 
We should sort the current validator set by valID and do a binary search of validator Index. 
Sorting is extra time complexity. 

Considering CurrentValSet count is less, sorting should be fine i think.

Reply:
@mankenavenkatesh  Validators in CurrentValSet will be already sorted as signer as far as I know. Can you check?